File: trabalhos 1 ANO/aluno.js
//criando a classe aluno
class aluno {
    constructor(numero, nome, idade, p1, p2){
        this.numero = numero
        this.nome = nome
        this.idade = idade
        this.p1 = p1
        this.p2 = p2
    }

    notaFinal = () => {
        return (this.p1 + this.p2) / 2
    }

    mostra = () => {
        console.log(`Numero: ${this.numero}, Nome: ${this.nome}, Idade: ${this.idade}, Nota final: ${this.notaFinal()},`)
    }

    passou = () => {
        if(this.notaFinal() > 6){
            return `${this.nome} foi aprovado`
        }
        else {
            return `${this.nome} foi reprovado`
        }
    }
}

//criando objetos
let aluno1 = new aluno(111, "Lucas", 20, 6, 8)
aluno1.mostra()
console.log(aluno1.passou())

let a = Number(prompt(`Informe o numero do aluno 2`))
let b = prompt(`Informe o nome do aluno 2`)
let c = Number(prompt(`Informe a idade do aluno 2`))
let d = Number(prompt(`Informe a nota 1 do aluno 2`))
let e = Number(prompt(`Informe a nota 2 do aluno 2`))
let aluno2 = new aluno(a, b, c, d, e)
aluno2.mostra()
console.log(aluno2.passou())

File: NETBEANS_PROJECTS_JV/Lista1/src/main/java/br/com/unifacef/lista1/ex1/Aluno.java
//Exercício 1: 
//•	Crie uma classe em Java chamada Aluno contendo as variáveis de instância numeroAluno (int), nome (String), idade (int), p1 (float) e p2 (float), 
//•	Crie os construtores 
//•	Crie os métodos notaFinal() - que calcula e retorna a média final do aluno - , e dadosAluno() - que retorna os valores de numeroAluno, nome e idade. 
//•	Criar uma classe TestaAluno com o método main() para instanciação da classe Aluno (atribuição das variáveis de instância com quaisquer valores e
//   chamada dos métodos notaFinal() e dadosAluno()).  
//•	Adicionar o método passou() à classe Aluno a fim de verificar se o aluno passou. Chame o método passou() na classe TestaAluno a fim de testa-lo.

package br.com.unifacef.lista1.ex1;

public class Aluno {
    //variaveis fortemente tipadas
    public int nroAluno, idade;
    public String nome;
    public float p1, p2;
    
    // na classe TestaAluno : Aluno obj1 = new Aluno()
    // construtor sem parametros
    public Aluno(){
    }
    
    public Aluno(int nroAluno, int idade, String nome, float p1, float p2){
        this.nroAluno = nroAluno;
        this.idade = idade;
        this.nome = nome;     
        this.p1 = p1;
        this.p2 = p2;
    }

    public float notaFinal(){
        return (this.p1 + this.p2) / 2;
    }
    //  método que retorna o valor
    //      System.out.println(obj1.dadosAluno())
    //      System.out.println(obj2.dadosAluno())
    //  método retorna a nota final
    //      Aluno obj1 = new Aluno(1, 20, "João", 4, 6)
    //      float aux = obj1.notaFinal()
    //      Aluno obj2 = new Aluno(2, 18, "Pedro", 8.5f, 6)
    //      float aux2 = obj2.notaFinal()
    
    // String é o tipo de retorno
    public String dadosAluno(){
        return "Nro do aluno : " + this.nroAluno + " Nome : " + this.nome + " Idade : " + this.idade;
    }
    
    // void pois o método não tem return
    public void passou(){
        float aux = this.notaFinal();
        if (aux >=6){
            System.out.println("Aprovado com média : " +aux );
        }else{
            System.out.println("Reprovado com média : " +aux );
        }
    }
    
    
}
